File: lib/features/map/domain/models/station.dart
import 'package:equatable/equatable.dart';
import 'package:station_reach/features/map/domain/enums/transit_mode.dart';

/// {@template station}
/// A transit station
/// {@endtemplate}
class Station extends Equatable {
  /// {@macro station}
  const Station({
    required this.id,
    required this.name,
    required this.latitude,
    required this.longitude,
    required this.modes,
    this.countryCode,
    this.area,
  });

  final String id;
  final String name;
  final double latitude;
  final double longitude;
  final List<TransitMode> modes;
  final String? countryCode;

  /// The name of the area of the station
  final String? area;

  @override
  List<Object?> get props => [
    id,
    name,
    latitude,
    longitude,
    modes,
    countryCode,
    area,
  ];
}

File: src/features/map/domain/models/station.ts
import type { TransitMode } from './transit-mode';

export interface Station {
  readonly id: string;
  readonly name: string;
  readonly latitude: number;
  readonly longitude: number;
  readonly modes: readonly TransitMode[];
  /** ISO 3166-1 country code, absent when the search result has none. */
  readonly countryCode?: string;
  /**
   * Short place name shown beside the station name, e.g. its city or
   * district, absent when none is available.
   */
  readonly area?: string;
}

export interface Stop extends Station {
  /**
   * Minutes from the departure's scheduled time at the origin. Zero for the
   * origin stop itself.
   */
  readonly durationMinutes: number;
}
